.evergreen/check-rustdoc.sh

```diff
@@ -5,4 +5,5 @@ set -o errexit
 . ~/.cargo/env
 cargo +nightly rustdoc -- -D warnings --cfg docsrs
 cargo +nightly rustdoc --no-default-features --features async-std-runtime -- -D warnings --cfg docsrs
-cargo +nightly rustdoc --no-default-features --features sync -- -D warnings --cfg docsrs
+cargo +nightly rustdoc --features sync -- -D warnings --cfg docsrs
```

Comment:
I think we still need `--no-default-features` here for the async-std sync variant. It looks like the `check-rustdoc.sh` test is failing as a result.

Reply:
oh whoops, good catch